### PR TITLE
Add Domain Connect template for Valimail

### PR DESCRIPTION
**Pull request by BrianSigafoos**

### CODE REVIEW

File: valimail.com.valimail-authenticate.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      "ttl": 3600
+    },
+    {
+      "type": "NS",
+      "host": "_bimi",
+      "pointsTo": "ns.vali.email",
+      "ttl": 3600
+    },
+    {
+      "type": "TXT",
```
Comment (pawel-kow) [inline valimail.com.valimail-authenticate.json:30]: For SPF please use SPFM record type.
Plain TXT record, especially with conflict mode, makes the template "own" the domains SPF, what goes beyond a scope of mail service.

File: valimail.com.valimail-authenticate.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "records": [
+      {
+      "type": "NS",
+      "host": "_dmarc",
+      "pointsTo": "ns.vali.email",
+      "ttl": 3600
+    },
+    {
+      "type": "NS",
+      "host": "_domainkey",
```
Comment (pawel-kow) [inline valimail.com.valimail-authenticate.json:19]: This one would not allow any other service to use DKIM. 
The record has a structure <key>._domainkey exactly to allow multiple keys per domain.
Also, not every DNS provider supports NS record.
Maybe better to work with CNAME on variable-based host?

File: valimail.com.valimail-authenticate.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      "pointsTo": "ns.vali.email",
+      "ttl": 3600
+    },
+    {
+      "type": "NS",
+      "host": "_bimi",
```
Comment (pawel-kow) [inline valimail.com.valimail-authenticate.json:25]: same comment as for _domainkey, this record would block more than one key.

File: valimail.com.valimail-authenticate.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  "serviceId": "valimail-authenticate",
+  "serviceName": "Valimail Authenticate",
+  "version": 1,
+  "logoUrl": "https://valimail-logos.s3.amazonaws.com/valimail-logo-primary.svg",
+  "description": "Enable Valimail Authenticate on a domain",
+  "variableDescription": "%spftxt% is the SPF record",
+  "records": [
+      {
+      "type": "NS",
+      "host": "_dmarc",
```
Comment (pawel-kow) [inline valimail.com.valimail-authenticate.json:13]: Generally we discourage non-conditional usage of DMARC, as it applies to the whole domain, not only to one service.

**Review by pawel-kow — CHANGES_REQUESTED** [review]

File: valimail.com.valimail-authenticate.json (hunk 4 above)
Comment (AntiFreeze) [reply valimail.com.valimail-authenticate.json:13]: Hi Pawel, thanks for the feedback. I think there's a general misunderstanding here, and I'd like to attempt to clarify what we do, and exactly what this template is asking for:

Our customers do NS delegations of _dmarc, _domainkey, and _bimi, and create a special SPF TXT record, and then our app provides management and maintenance of everything at or under those labels. We're able to auto-import the existing data through other means, so these actions are not destructive.

So yes, we fully agree that DMARC is about the entire domain. That's why the above delegations are what we ask for-- we're then able to manage and authenticate each individual service without our customers ever needing to make manual DNS changes again.

We do this all the time, for instance, check out NS _dmarc.uber.com or _domainkey.epa.gov or _bimi.hbo.com (or the v=spf1 TXT record on each of those domains).

With domain connect, our customers won't even need to make 3 NS delegations and 1 TXT record update, they'll just need to press the button they've been asking for.

Let me know if that makes sense, we're happy to clarify.

File: valimail.com.valimail-authenticate.json (hunk 2 above)
Comment (AntiFreeze) [reply valimail.com.valimail-authenticate.json:19]: As described above, this is what we're attempting to achieve with this template. For providers that do not support NS records, we do fall back to CNAME, but that's out of the scope of this template.

File: valimail.com.valimail-authenticate.json (hunk 3 above)
Comment (AntiFreeze) [reply valimail.com.valimail-authenticate.json:25]: Understood, as with DKIM we manage all selectors under the label. (Note: I'm the author of the BIMI spec: https://tools.ietf.org/html/draft-blank-ietf-bimi-01)

File: valimail.com.valimail-authenticate.json (hunk 1 above)
Comment (AntiFreeze) [reply valimail.com.valimail-authenticate.json:30]: The SPF RR type has been deprecated, and must not be used, see:
- https://tools.ietf.org/html/rfc6686
- https://tools.ietf.org/html/rfc7208 (section 14.1)
- https://www.m3aawg.org/documents/en/m3aawg-best-practices-for-managing-spf-records (section 3.2.1)

File: valimail.com.valimail-authenticate.json (hunk 4 above)
Comment (pawel-kow) [reply valimail.com.valimail-authenticate.json:13]: Thanks, this explanation was quite helpful.
Looks like your service is similar to [FraudMarc](https://github.com/Domain-Connect/Templates/blob/master/fraudmarc.com.mail.json).

Then I don't have any further questions actually and I can approve this template.
@arnoldblinn would you take a look as well?

**Review by pawel-kow — APPROVED** [review]

**Comment by BrianSigafoos** [comment]
@pawel-kow @arnoldblinn anything I can do to help with the pending review?

**Comment by arnoldblinn** [comment]
This looks fine to me.

**Review by arnoldblinn — APPROVED** [review]

**Comment by pawel-kow** [comment]
@BrianSigafoos all OK. Please pull the changes from master to this PR branch so we can merge.

**Comment by BrianSigafoos** [comment]
@pawel-kow thank you. Done.

**Comment by pawel-kow** [comment]
Merged. Please let us know if you want to get listed on domainconnect.org/service-providers/

**Comment by BrianSigafoos** [comment]
@pawel-kow yes please 🙏 
We're glad for it to look like the others:
"For DNS Providers interested in working with Valimail on Domain Connect contact domainconnect@valimail.com"
Also can use the same logo as this PR: https://valimail-logos.s3.amazonaws.com/valimail-logo-primary.svg

**Pull request merged**